sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -911,7 +911,7 @@ class CastChecks extends ExprChecks {
   val calendarChecks: TypeSig = none
   val sparkCalendarSig: TypeSig = CALENDAR + STRING
 
-  val arrayChecks: TypeSig = none
+  val arrayChecks: TypeSig = ARRAY.nested(DOUBLE)
```

Comment:
This is saying that if you are casting an array of anything to an Array of doubles we are good if Spark also supports it. We need to have more code to either make the checks match what the code supports or make the code more generic so it can support what support what the checks say is allowed.